docs/core/integration-quality-scale/rules/iqs028.md

```diff
@@ -0,0 +1,43 @@
+---
+title: "IQS028 - If making HTTP requests, support passing in websession"
+---
+
+IQS028 is as follows:
+
+If making HTTP requests, support passing in websession
+
+## Reasoning
+
+Since a lot of devices and services are connected to via HTTP, the amount of websessions that are active can be a lot.
+To improve the efficiency of those websessions, it is recommended to support passing in a websession.
+These sessions are tweaked towards being used in Home Assistant and is more efficient.
+
+Home Assistants supports this for `aiohttp` and `httpx`.
+
+## Example implementation
+
+In the example below, a `aiohttp` session is passed in to the client.
+The equivalent for `httpx` would be `get_async_client`.
+
+```python
+async def async_setup_entry(hass: HomeAssistant, entry: MyConfigEntry) -> bool:
+    """Set up my integration from a config entry."""
+
+    client = MyClient(entry.data[CONF_HOST], async_get_clientsession(hass))
+```
+
+:::info
+There are cases where you might not want a shared session, for example when cookies are used.
+In that case, you can create a new session using `async_create_clientsession` for `aiohttp` and `create_async_httpx_client` for `httpx`.
+:::
+
+## Additional resources
+
+
+## Exceptions
+
+There are no exceptions to this rule.
```

Comment:
Do we know any exceptions to this?

Reply:
Yup, in case `aiohttp` and `httpx` aren't used? :)

Reply:
Okay so to combine this with the async dependency rule.

If someone created a library that worked with requests, and uses the `hass.loop` as parameter to create their own executor jobs in the library to make the API we use async. (I think this is a realistic use case).

Do we still allow them for both rules? Because in that case the dependency is "async" (not optimal async, but we talk async to the library) thus that rule would be checked, and this rule would not be checked since its exempt.

Reply:
We shouldn't allow such a library, I think. That's a very bad pattern.

Reply:
I agree. But what is the best way to enforce that? In what case would one be making HTTP calls in async context and not be using `aiohttp` or `httpx`

Reply:
To meet this rule, the integration should use an async library that either uses aiohttp or httpx for http requests. I don't think we need an exception for this.

If the library doesn't do http requests the rule doesn't apply.

Reply:
That would be the exception I guess; If the integration doesn't do HTTP requests.

Reply:
Then we should move the exception from the title to the exceptions